content/articles/how-to-add-support-for-spatial-data-in-your-database-using-geoalchemy-and-spatialite/index.md

```diff
@@ -0,0 +1,397 @@
+- Python3 Installed
+- Basic Flask Knowledge
+
+We'll start by creating a folder called `server` and in this folder, we'll add a new folder called `core` and three other files called `wsgi.py`, `models.py` and `config.py`. Inside the `core` folder, add two new files called: `__init__.py`, and `views.py`. Once you're done your `server` folder should have a structure like this one below:
+
+```bash
+-server/
+    -core/
+        -__init__.py
+        -views.py
+    -models.py
+    -wsgi.py
+    -config.py
+```
+
+#### Installing Dependencies
+From the command prompt/terminal, navigate to the server folder we just created and create a new virtual environment for the project, using `virtualenv`. If you don't have `virtualenv` installed you can install via `pip` as follows:
+
+```bash
+ pip install virtualenv
+```
+
+If you already have it installed, you can skip that step and install the libraries we need as follows:
+
+```bash
+pip install flask flask-sqlalchemy flask-migrate geoalchemy2
+```
+
+#### Creating and configuring the Flask server
+In the `__init__.py` file in the **core/** folder we will add the following lines of code to create an application factory. This is a function that we can call to help with creating and configuring out flask application - this is great, because it can take arguments that we need to create an application at any point when we call the function.
+
+```python
+from flask import Flask
+from flask_sqlalchemy import SQLAlchemy
+from flask_migrate import Migrate
+from config import config_
+
+# initialize extensions
+db = SQLAlchemy()
+migrate = Migrate()
+
+# app factory
+def create_app(config_name):
+    # create app instance
+    app = Flask(__name__)
+
+    # configure app
+    app.config.from_object(config_[config_name])
+
+    # initialize extensions
+    db.init_app(app)
+    migrate.init_app(app, db)
+
+    return app
+
+```
+Let's add some configuration options that our flask app can use, in the `config.py` file we created earlier. So navigate to that file and add the following lines of code to add the configuration options.
+
+```python
+import os
+
+base_dir = os.path.abspath(os.getcwd())
+
+class Config:
+    SQLALCHEMY_TRACK_MODIFICATIONS = False
+    SECRET_KEY = os.getenv('app-secret')
+    # other generic configuration
+    # ..options
+
+class DevConfig(Config):
+    SQLALCHEMY_DATABASE_URI = f"sqlite:///{base_dir}/devdb.sqlite"
+    DEBUG = True
+    FLASK_ENV = 'development'
+
+class ProdConfig(Config):
+    SQLALCHEMY_DATABASE_URI = os.getenv('DATABASE_URI')
+    FLASK_ENV = 'production'
+
+config_ = {
+    'default': DevConfig,
+    'development': DevConfig,
+    'production': ProdConfig
+}
+```
+The `_config` variable in the last line holds a  dictionary that maps each config option to their names, so we can use the `config's name` to reference them when we need to add the config to our flask application. This is what we did in the `__init__.py` file, in the app factory, as shown below:
```

Comment:
`_config`

Is this the variable you used in the above code snippet?

Reply:
yes i misspelled it here should be "config_"